gtsfm/densify/patchmatchnet_data.py

```diff
@@ -0,0 +1,267 @@
+"""Interface class from GtsfmData to PatchmatchNetData
+
+Authors: Ren Liu
+"""
+from typing import Any, Dict, List, Tuple
+
+import cv2
+import numpy as np
+from torch.utils.data import Dataset
+
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.common.image import Image
+from gtsfm.densify.mvs_math import piecewise_gaussian
+
+NUM_PATCHMATCHNET_STAGES = 4
+
+MIN_DEPTH_PERCENTILE = 1
+MAX_DEPTH_PERCENTILE = 99
+
+
+class PatchmatchNetData(Dataset):
+    """PatchmatchNetData class for PatchmatchNet. It contains the interface from GtsfmData.
+    Ref: Wang et al. https://github.com/FangjinhuaWang/PatchmatchNet/blob/main/datasets/dtu_yao_eval.py
+    """
+
+    def __init__(self, images: Dict[int, Image], sfm_result: GtsfmData, num_views: int = 5) -> None:
+        """Cache image and camera pose metadata for PatchmatchNet inference.
+
+        Args:
+            images: input images (H, W, C) to GTSFM
+            sfm_result: sfm results calculated by GTSFM
+            num_views: number of views, containing 1 reference view and (num_views-1) source views
```

Comment:
What is the expected behavior of num_views is 5, but only 3 images are provided?

Reply:
I think generally there will be more than 5 images, or the densify process is likely to fail. I add 
```python
# Force the number of views to be equal to or larger than the number of images
num_views = min(num_views, len(images))
```
to it

Reply:
got it, makes sense. could you document that in the docstring also?